File: sdk-python/examples/features/auto_paginate.py
"""Iterate every active feature in the catalog, transparently fetching each
page as the previous one drains.

Run with: python examples/features/auto_paginate.py
"""

from __future__ import annotations

from threecommon import ThreeCommon
from threecommon.features import ListParams


def main() -> None:
    count = 0
    quantity = 0
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        for feature in client.features.list_auto_paginate(ListParams(active=True)):
            count += 1
            if feature.type == "quantity":
                quantity += 1

    print(f"iterated {count} active features")
    print(f"{quantity} are quantity-typed")


if __name__ == "__main__":
    main()

File: sdk-node/examples/features/auto-paginate.ts
/**
 * Iterate every active feature in the catalog, transparently fetching each
 * page as the previous one drains.
 *
 * Run:
 *   npx tsx examples/features/auto-paginate.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

let count = 0
let quantity = 0
for await (const feature of client.features.listAutoPaginate({ active: true })) {
  count += 1
  if (feature.type === 'quantity') quantity += 1
}

console.log(`iterated ${String(count)} active features`)
console.log(`${String(quantity)} are quantity-typed`)
